synapse/lib/remcycle.py

```diff
@@ -0,0 +1,846 @@
+                apis_to_remove.append(api_name)
+
+        for api_name in apis_to_remove:
+            self._deregister_api(name=api_name)
+
+        self.fire('hypnos:register:namespace:del', namespace=namespace)
+
+    def _deregister_api(self, name):
+        if name not in self.apis:
+            raise NoSuchName('API name not registered.')
+
+        self.apis.pop(name, None)
+
+        funclist = self._api_ingests.pop(name, [])
+        for action_name, ingest_func, gest_glue in funclist:
+            self.off(name, gest_glue)
+            self.core.off(action_name, ingest_func)
+
+        self.fire('hypnos:register:api:del', api=name)
+
+    def getApiNyx(self, name):
+        '''
+        Get the Nyx object corresponding to a given API name.
+
+        :param name: Name of the API to get the object for.
+        :return: A Nyx object.
+        :raises: A NoSuchName error if the requested name does not exist.
+        '''
+        nyx = self.apis.get(name)
+        if not nyx:
+            log.error('No name registered with Hypnos as %s', name)
+            raise NoSuchName
+        return nyx
+
+    @staticmethod
+    def _flatten_response(resp):
+        '''Flatten the Tornado HTTPResponse object to a dictionary.'''
+        resp_dict = {
+            'request': {'url': resp.request.url,
+                        'headers': dict(resp.request.headers)}
+        }
+        error = resp.error
+        if error and not isinstance(error, t_http.HTTPError):
+            return resp_dict
+        resp_dict.update({'code': resp.code,
+                          'raw_body': resp.body,
+                          'headers': dict(resp.headers),
+                          'effective_url': resp.effective_url, })
+        if not resp.body:
+            return resp_dict
+        ct = resp.headers.get('Content-Type', 'text/plain')
+        ct_type, ct_params = cgi.parse_header(ct)
+        charset = ct_params.get('charset', 'utf-8').lower()
+        try:
+            resp_dict['data'] = resp_dict.get('raw_body').decode(charset)
+        except Exception as e:
+            log.exception('Failed to decode a raw body in a response object.')
+            return resp_dict
+        # Handle known content types and put them in the 'data' key
+        # we can add support for additional data types as needed.
+        if ct_type.lower() == 'application/json':
+            resp_dict['data'] = json.loads(resp_dict.get('data'))
+        return resp_dict
+
+    def _resp_fail_wrapper(self, f):
+        '''Decorator for wrapping callback functions to check for exception information.'''
+
+        def check_job_fail(*fargs, **fkwargs):
+            _excinfo = fkwargs.get('excinfo')
+            if _excinfo:
+                _jid = fkwargs.get('jid')
+                self.boss.err(jid=_jid, **_excinfo)
+            else:
+                f(*fargs, **fkwargs)
+
+        return check_job_fail
+
+    def fireApi(self, name, *args, **kwargs):
```

Comment:
Renamed everything in Hypnos to be "Web" as a prefix, since that is short and easy to type.